### PR TITLE
Sir 336 create random review page

### PR DESCRIPTION
**Pull request by Nicholassully**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "license": "MIT",
   "scripts": {
     "build": "webpack",
-    "cypress": "cypress open"
+    "cypress": "cypress open",
+    "test-sirius": "cd internal/sirius && go test",
```
Comment (hawx) [inline package.json:8]: Do we need these when it is simply `go test ./internal/sirius` or `go test ./internal/server` (or `go test ./...` to run all)?

File: yarn-error.log [hunk 2]
```diff
@@ -0,0 +1,5265 @@
+Arguments:
```
Comment (hawx) [inline yarn-error.log:1]: Remove this and add to `.gitignore`?

File: README.md [hunk 3]
```diff
@@ -49,14 +56,13 @@ This will run the Go unit tests. It relies on `pact` being available on your
 provide a stub service for the Cypress tests.
 
 ```
-docker-compose -f docker/docker-compose.cypress.yml up -d --build
+docker-compose -f docker/docker-compose.cypress.yml up -d --build --force-recreate
```
Comment (hawx) [inline README.md:59]: Is `--force-recreate` necessary? If you need to restart containers then it is better to use `down` then `up`

**Review by hawx — COMMENTED** [review]
Also looks like you need to `go fmt` everything

File: cypress/fixtures/example.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (denialanderror) [inline cypress/fixtures/example.json:1]: Does this need to be committed?

File: cypress/integration/edit_lay_percentage.spec.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+describe("Edit lay percentage", () => {
+    beforeEach(() => {
+        cy.setCookie("Other", "other");
+        cy.setCookie("XSRF-TOKEN", "abcde");
+        cy.visit("/random-reviews/edit/lay-percentage");
+    });
+
+    it("allows me to change the lay percentage value", () => {
```
Comment (denialanderror) [inline cypress/integration/edit_lay_percentage.spec.js:8]: This doesn't appear to describe what is being tested. You are testing that an error message displays if you input the incorrect value.

File: package.json (hunk 1 above)
Comment (Nicholassully) [reply package.json]: done

File: yarn-error.log (hunk 2 above)
Comment (Nicholassully) [reply yarn-error.log]: done

File: README.md (hunk 3 above)
Comment (Nicholassully) [reply README.md]: done

File: cypress/integration/random_reviews.spec.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+        cy.get(".hook-layPercentageRow").each(($el, index) => {
+            cy.wrap($el).within(() => {
+                cy.get(".hook-layPercentageKey").contains(expected[index][0]);
+                cy.get(".hook-layPercentageValue").contains(expected[index][1]);
+            });
+        });
+    });
+
+    it("allows me to change the lay percentage", () => {
```
Comment (denialanderror) [inline cypress/integration/random_reviews.spec.js:22]: These don't test what they describe either. They are asserting on the presence of the button, not that you can use it.

File: internal/server/edit_lay_percentage.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+type editLayPercentageVars struct {
+	Path                string
+	XSRFToken           string
+	LayPercentage       string
+	ReviewCycle         string
+	Errors              sirius.ValidationErrors
+	Error               string
+}
+
+func editLayPercentage(client EditLayPercentageClient, tmpl Template) Handler {
```
Comment (denialanderror) [inline internal/server/edit_lay_percentage.go:24]: Don't know if it is an issue with github displaying it but the formatting is really off in this function

File: internal/sirius/add_team.go [hunk 8]
```diff
@@ -41,6 +41,7 @@ func (c *Client) AddTeam(ctx Context, name, teamType, phone, email string) (int,
 		return 0, ErrUnauthorized
 	}
 
+
```
Comment (denialanderror) [inline internal/sirius/add_team.go:44]: unnecessary line break

File: internal/sirius/search_users.go [hunk 9]
```diff
@@ -67,23 +64,21 @@ func (c *Client) SearchUsers(ctx Context, search string) ([]User, error) {
 		return nil, newStatusError(resp)
 	}
 
-	var v apiUserList
+	var v []apiUser
```
Comment (denialanderror) [inline internal/sirius/search_users.go]: Why move from a more descriptive name to a less descriptive one?

**Review by denialanderror — CHANGES_REQUESTED** [review]
I don't know what's happened to the formatting but can you try running go fmt over it.

There are also a lot of changes that don't appear related to creating random reviews (e.g. all the changes to the v1 endpoints). It would have been easier to review if these were part of a separate PR.

File: internal/server/edit_lay_percentage.go (hunk 7 above)
Comment (gurps-singh) [reply internal/server/edit_lay_percentage.go:24]: go fmt ran

File: cypress/integration/edit_lay_percentage.spec.js (hunk 5 above)
Comment (gurps-singh) [reply cypress/integration/edit_lay_percentage.spec.js]: done

File: cypress/integration/random_reviews.spec.js (hunk 6 above)
Comment (gurps-singh) [reply cypress/integration/random_reviews.spec.js]: done

File: internal/sirius/add_team.go (hunk 8 above)
Comment (gurps-singh) [reply internal/sirius/add_team.go]: done

File: internal/sirius/search_users.go (hunk 9 above)
Comment (gurps-singh) [reply internal/sirius/search_users.go]: change made in a previous PR and rebased into ours

File: cypress/integration/edit_lay_percentage.spec.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+describe("Edit lay percentage", () => {
+    beforeEach(() => {
+        cy.setCookie("Other", "other");
+        cy.setCookie("XSRF-TOKEN", "abcde");
+        cy.visit("/random-reviews/edit/lay-percentage");
+    });
+
+    it("allows me to change the lay percentage value and throws an error after inputting the incorrect value", () => {
```
Comment (denialanderror) [inline cypress/integration/edit_lay_percentage.spec.js:8]: This still isn't describing what you are testing. "throws an error after inputting the incorrect value" is fine but you aren't changing the lay percentage value.

**Review by denialanderror — APPROVED** [review]
One comment about test naming but happy to approve

**Review by denialanderror — APPROVED** [review]

File: cypress/integration/edit_lay_percentage.spec.js (hunk 10 above)
Comment (gurps-singh) [reply cypress/integration/edit_lay_percentage.spec.js]: Doh! Done

**Pull request merged**